jetty-server/src/main/java/org/eclipse/jetty/server/HostHeaderCustomizer.java

```diff
@@ -55,14 +58,29 @@ public HostHeaderCustomizer(String serverName)
      */
     public HostHeaderCustomizer(String serverName, int serverPort)
     {
-        this.serverName = Objects.requireNonNull(serverName);
-        this.serverPort = serverPort;
+        String host = Objects.requireNonNull(serverName);
+        if (serverPort > HttpURI.NO_PORT)
+            host += ":" + serverPort;
+        hostValue = host;
     }
 
     @Override
     public void customize(Connector connector, HttpConfiguration channelConfig, Request request)
     {
-        if (request.getHeader("Host") == null)
-            request.setAuthority(serverName, serverPort);  // TODO set the field as well?
+        String hostHeaderValue = request.getHeader("Host");
+
+        // No Host Header
+        if (request.getHttpVersion().getVersion() != HttpVersion.HTTP_1_1.getVersion() &&
```

Comment:
This can cause NPE, isn't it? request.getHttpVersion() can return null.

Reply:
The request shouldn't reach this customizer then, as I would expect the request to fail well before the customizers have been called.